x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/utils/helpers.tsx

```diff
@@ -62,15 +60,19 @@ export const filterIndexPatterns = (
 
       return {
         ...patterns,
-        fields: patterns.fields.filter(({ name }) => osFilterForEndpoint(name)),
-      };
+        fields:
+          patterns != null ? patterns.fields.filter(({ name }) => osFilterForEndpoint(name)) : [],
+      } as Omit<DataView, 'fields'> & { fields: FieldSpec[] };
```

Comment:
we should avoid doing any casts like this and likely delete this new hybrid type altogether. Everything in security solution and the dependent plugins should work with properties from the DataViewSpec type, and serialize back to an instance of DataView class 1) as rarely as possible, only when something completely external expects it 2) at the last possible moment, as the size of the class is much bigger/unserializable in redux where a lot of this code touches